types/angular/index.d.ts

```diff
@@ -1029,8 +1029,7 @@ declare namespace angular {
          *
          * @param promises A hash of promises.
          */
-        all(promises: { [id: string]: IPromise<any>; }): IPromise<{ [id: string]: any; }>;
-        all<T extends {}>(promises: { [id: string]: IPromise<any>; }): IPromise<T>;
+        all<T extends {}>(promises: { [K in keyof T]: IPromise<T[K]>; }): IPromise<T>;
```

Comment:
Can you add or modify tests to show how this change alters what can be done with the new signature for `all`? 

I found that the old version detects objects with non-Promise-wrapped properties, but of course returns a much worse type. The new version returns a nice type, but doesn't catch incorrect objects with non-Promise-wrapped properties. You just get a bogus type: `IPromise<{}>`. I want to make sure that you're aware of the tradeoff and are OK with it.